lib/server_manager.go

```diff
@@ -0,0 +1,158 @@
+package lib
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/netip"
+	"sync"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/fatih/color"
+	"golang.zx2c4.com/wireguard/wgctrl"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+type ServerInfo struct {
+	i      uint16
+	cancel context.CancelFunc
+}
+
+// ServerManager handles creating and terminating servers on ips
```

Comment:
Maybe good to mention that ServerManager is not threadsafe for concurrent access in a comment.